### PR TITLE
Fix inifinite retry issue in telemetry upload

### PR DESCRIPTION
**Pull request by troydai**
Related: https://github.com/Microsoft/ApplicationInsights-Python/issues/60

Fixes: #3177 #4649 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/telemetry_upload.py [hunk 1]
```diff
@@ -37,7 +78,7 @@ def upload(data_to_save):
         data_to_save = json.loads(data_to_save.replace("'", '"'))
     except Exception as err:  # pylint: disable=broad-except
         if in_diagnostic_mode():
-            sys.stdout.write('{}/n'.format(str(err)))
+            sys.stdout.write('ERROR: {}/n'.format(str(err)))
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/telemetry_upload.py]: Why 'Error'? If it's supposed to be there, it's fine with me.

**Comment by derekbekoe** [comment]
Consider modifying history?

File: src/azure-cli-core/azure/cli/core/telemetry_upload.py (hunk 1 above)
Comment (troydai) [reply src/azure-cli-core/azure/cli/core/telemetry_upload.py]: This is just for the sake of clear output.

**Comment by troydai** [comment]
Will update the history.

**Comment by tjprescott** [comment]
Is there a happy medium between 0 retries and infinite?  It just seems that some kind of retry logic is just good practice for networked applications.

**Comment by troydai** [comment]
@tjprescott this is for telemetry only. It won't impact command's network communication.

**Pull request merged**